soc/xtensa/intel_adsp/tools/cavstool.py

```diff
@@ -387,18 +406,25 @@ def load_firmware(fw_file):
 
     # Start DSP.  Host needs to provide power to all cores on 1.5
     # (which also starts them) and 1.8 (merely gates power, DSP also
-    # has to set PWRCTL).  The bits for cores other than 0 are ignored
-    # on 2.5 where the DSP has full control.
+    # has to set PWRCTL). On 2.5 where the DSP has full control,
+    # and only core 0 is set.
     log.info(f"Starting DSP, ADSPCS = 0x{dsp.ADSPCS:x}")
-    dsp.ADSPCS = 0xff0000 if not cavs25 else 0x01fefe
-    while (dsp.ADSPCS & 0x1000000) == 0: pass
+    dsp.ADSPCS = mask(SPA)
+    while (dsp.ADSPCS & mask(CPA)) == 0: pass
+
+    log.info(f"Unresetting DSP cores, ADSPCS = 0x{dsp.ADSPCS:x}")
+    dsp.ADSPCS &= ~mask(CRST)
+    while (dsp.ADSPCS & 1) != 0: pass
+
+    log.info(f"Running DSP cores, ADSPCS = 0x{dsp.ADSPCS:x}")
+    dsp.ADSPCS &= ~mask(CSTALL)
 
-    # Wait for the ROM to boot and signal it's ready.  This short
+    # Wait for the ROM to boot and signal it's ready.  This not so short
     # sleep seems to be needed; if we're banging on the memory window
     # during initial boot (before/while the window control registers
     # are configured?) the DSP hardware will hang fairly reliably.
-    log.info("Wait for ROM startup")
-    time.sleep(0.1)
+    log.info(f"Wait for ROM startup, ADSPCS = 0x{dsp.ADSPCS:x}")
+    time.sleep(1)
```

Comment:
It really needs a full second?  That's frustrating.  That is starting to get to the point where it impacts test throughput.

Reply:
I can certainly recheck, but 0.5 wasn't enough, so my next step was the second.

Reply:
Second is a lot indeed, but Linux driver does have a 500msec init timeout, so in the same ballpark.